src/Umbraco.Web/PublishedCache/NuCache/DataSource/SerializerBase.cs

```diff
@@ -154,8 +172,46 @@ protected void WriteObject(object value, Stream stream)
             }
             else if (value is LazyCompressedString lazyCompressedString)
             {
-                PrimitiveSerializer.Char.WriteTo(PrefixCompressedStringByteArray, stream);
-                PrimitiveSerializer.Bytes.WriteTo(lazyCompressedString.GetBytes(), stream);
+                if (lazyCompressedString.IsDecompressed())
```

Comment:
Not sure I understand this one. If ToString() is called on the LazyCompressedString it will be decompressed and because of that it will write it to the cache file as a string instead of bytes? When is the logic that determines this?

Reply:
If the type is LazyCompressedString , PrefixCompressedStringByteArray is the prefix. If the string has been decompressed, bytes are null. So it has to be written as a string.

Reply:
Will that ever happen though? If we do that then there's going to be a problem reading it because we are always reading as bytes.

Reply:
It happens. That's how I discovered this.

Reply:
Righto, will have to get some testing done to see where this is all at. Been too long in between reviews/work!

Reply:
The only way that GetBytes() can get called multiple times and then throw an exception is during debugging if you inspect the value of LazyCompressedString since internally that will call ToString() and the bytes will be read then resulting in an exception. 

We cannot do this logic here by checking if it's decompressed because that will end up putting things into the cache file in inconsistent ways.

I'll change how LazyCompressedString works and add a DebuggerDisplay to prevent this from occurring.